Support centered settings

yoyo837 opened the pull request:
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style. `windows`
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

weidapao committed:
Update .codecov.yml

adybionka committed:
fix: table onChange event is fired every time when close filterDropdown

Close #11164

weidapao committed:
删除手误

jljsj33 committed:
update drawer

chenshuai2144 committed:
style: Adjust the edge shadow

chenshuai2144 committed:
test: change moment to render

jljsj33 committed:
update drawer test

chenshuai2144 committed:
test: Perfect test

afc163 committed:
Update index.en-US.md

Commit:
Strengthen Mention Component typings

afc163 committed:
site: update document title in home page

Tribute to material-ui.com and :wink:

afc163 committed:
site: styling API Tables

yoyo837 committed:
Support centered settings

yoyo837 committed:
Support centered settings

afc163 committed:
Update package.json

afc163 commented on components/modal/style/modal.less:
```diff
@@ -133,6 +134,42 @@
+.vertical-center-modal .ant-modal {
+  top: 0;
+}
+*/
```
这些代码没用到建议删掉。

afc163 commented on components/modal/style/modal.less:
```diff
@@ -133,6 +134,42 @@
   }
 }
 
+.vertical-center-modal {
```
建议改为 `.@{@dialog-prefix-cls}-centered`

afc163 commented on components/modal/index.zh-CN.md:
```diff
@@ -20,6 +20,7 @@ title: Modal
 | afterClose | Modal 完全关闭后的回调 | function | 无 |
 | bodyStyle | Modal body 样式 | object | {} |
 | cancelText | 取消按钮文字 | string | 取消 |
+| centered | 居中 Modal | Boolean | `false`
```
垂直居中展示 Modal

afc163 commented on components/modal/index.en-US.md:
```diff
@@ -62,6 +62,7 @@ The properties of the object are follows:
 | Property | Description | Type | Default |
 | -------- | ----------- | ---- | ------- |
 | cancelText | Text of the Cancel button | string | `Cancel` |
+| centered | Centered Modal | Boolean | `false`
```
加的位置和中文文档不一致。

afc163 commented on components/modal/style/modal.less:
```diff
@@ -133,6 +134,42 @@
+    &-body {
+      white-space: nowrap;
+      overflow-x: auto;
+    }
```
Why adding this?

yoyo837 replied:
The table in modal, `white-space:nowrap;` horizontal scrolling seems to experience better, or do you mean that it is better controlled by the table itself?

tangjinzhou committed:
fix: ButtonGroup style bug when have disabled button #9296 (#11321)

* fix: ButtonGroup style bug when have disabled button #9296

* Update mixin.less

zombieJ committed:
Bump 3.7.1 & update change log

chenshuai2144 committed:
remove snap in  destroyOnClose test

yoyo837 committed:
fix doc、css prefix、confirm centered modal supports

yoyo837 committed:
fix doc、css prefix、confirm centered modal supports

yoyo837 committed:
getDerivedStateFromProp return null if not change

A valid state object (or null) must be returned. You have returned undefined.

hongxuWei committed:
Fix the style of horizontal menu on dark theme

chenshuai2144 committed:
add wrapClassName and multi-level-drawer test

weidapao committed:
fix(Tooltip): fix text overflow the Tooltip

emersonlaurentino committed:
add headStyle to card component

Commit:
Fix types for Radio Group

zombieJ committed:
adjust change log

zombieJ committed:
update change log

afc163 committed:
Fix InputNumber overlap in fixed columns table

close #11408

ref #6289

afc163 committed:
Fix badge url

afc163 committed:
upgrade some devDeps

afc163 committed:
update snapshots

zombieJ committed:
move change log Drawer to first line

yesmeck committed:
Update .travis.yml

yesmeck committed:
test: mock React.createRef

yesmeck committed:
test: fix React.createContext mock

yesmeck committed:
test: fix react mock

yesmeck committed:
introduce circle ci

yesmeck committed:
Tweak circleci config

yesmeck committed:
chore: fix install react on circleci

yesmeck committed:
chore: tweak circleci

yesmeck committed:
chore: dry circleci config

yesmeck committed:
chore: persist whole workspace

yesmeck committed:
chore: fix job name

yesmeck committed:
chore: update circleci

afc163 committed:
Fix select menu item radius style

zongzi531 committed:
fix: badge type error.

yoyo837 committed:
Fix content without vertical alignment icon  when using vertical label mode; And description text-align: center;

yoyo837 committed:
progress-dot

yoyo837 committed:
docs

yoyo837 committed:
fix doc

Commit:
Update index.zh-CN.md (#11427)

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

yesmeck committed:
chore: remove circleci cache

paranoidjk committed:
feat(Avatar): expose onError (#11285)

* feat(Avatar): expose onError

* refactor: test case coverage

RaphaelChauveau committed:
Multiple row selection (#11406)

This PR intends to solve issue #11404 (Multiple row selection in table component).

The Chinese documentation is missing from the PR. Could someone please add it ?

yesmeck committed:
Merge branch 'master' into feature-3.8.0

afc163 committed:
Fix extra scrollbar style of Modal.confirm in Firefox, close #11432

yesmeck committed:
docs: add resizable column table demo

JribiBelhassen committed:
Change label in TreeData interface to title, since label is deprecated (#11442)

yesmeck committed:
fix: DatePicker can not change year/month when under control

React 16.4 call getDerivedStateFromProps on every render

close #11447
close #11440
close #11439
close #11416
close #11405

yesmeck committed:
chore: add tests for b9992f4

zombieJ committed:
make title & label as optional props in treeSelect

yesmeck committed:
Add 3.7.2 changelog (#11455)

yesmeck committed:
Bump 3.7.2

dependabot-support committed:
build(deps-dev): update stylelint requirement to 9.4.0

Updates the requirements on [stylelint](https://github.com/stylelint/stylelint) to permit the latest version.
- [Release notes](https://github.com/stylelint/stylelint/releases)
- [Changelog](https://github.com/stylelint/stylelint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stylelint/stylelint/commits/9.4.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

tangjinzhou committed:
feat: add onChange for Calendar

Commit:
Update Pagination Typescript definition to include "role" (#11474)

* Add data-, aria- and role prop types to match rc-pagination

* Remove data- and aria- definitions since they are allowed already

thisJJ committed:
Add id to Select props (#11189)

* Can set id to props on Select

* Update index.tsx

ilanus committed:
🐛 fix icons page 100% cpu usage

egfanboy committed:
fixed typo in `InputNumber` docs

405go committed:
fix: get label should use childrenKeyName for arrayTreeFilter when us… (#11311)

* fix: get label should use childrenKeyName for arrayTreeFilter when use customer fieldNames otherwise It can not get correct label array

* feat: add fieldNames test case in Cascader

* fix: cli fail

* fix: cil fail again

littleLane committed:
feat: add rowSelection.columnTitle to Table

yesmeck committed:
Add 3.7.3 changelog (#11494)

yesmeck committed:
Bump 3.7.3

yesmeck committed:
Add testUrl

https://github.com/jsdom/jsdom/issues/2304

yociduo committed:
export mouse enter and mouse leave event for datepicker

ilanus committed:
💬 fixing upload component typos and capitalization consistency for en-US

afc163 committed:
docs: Add instruction about cascader typo api

ilanus committed:
💬 fix tree component typos

ilanus committed:
💬 fix tabs component blank spaces

ilanus committed:
💬 update layout copyrights year and fix typos

ilanus committed:
💬 input component blank spaces remove

ilanus committed:
💬 misc typos and blank spaces fixes

ilanus committed:
✨ popover that can be hovered and clicked

ilanus committed:
🐛 update popover hover and click example order

yociduo committed:
add test case for date picker in tooltip

yociduo committed:
add timeout to wait trigger result

afc163 committed:
upgrade devDeps

afc163 committed:
chore DatePicker and TimePicker font style

afc163 committed:
Fix TimePanel jump issue of hovering, close #11460

afc163 committed:
Merge branch 'master' into feature-3.8.0

afc163 committed:
upgrade rc-tabs to 9.3.x

close #11261

https://github.com/react-component/tabs/pull/123

afc163 committed:
upgrade rc-select to 8.1.x

* support data-* attributes
* fix #11268

afc163 committed:
fix tabs test snapshots

afc163 committed:
demo: upgrade popover demo

afc163 committed:
fix card tabs demo

afc163 committed:
Merge branch 'master' into feature-3.8.0

afc163 committed:
upgrade rc-tabs and resolve propTypes warning

zombieJ committed:
update rc-tree

zombieJ committed:
Merge remote-tracking branch 'origin/feature-3.8.0' into tmp

zombieJ committed:
fix #11458. Adjust treeNode draggable css

afc163 committed:
docs: Add zhihu article link

yesmeck committed:
chore: add nightly build and ignore gh-pages

yesmeck committed:
chore: fix dir copy

afc163 committed:
Merge branch 'master' into feature-3.8.0

afc163 committed:
fix tabs ci in React 15

yoyo837 committed:
Merge branch 'feature-3.8.0' of https://github.com/yoyo837/ant-design into feature-3.8.0
